### ISSUE TITLE
Investigate JBrowse v2

### ISSUE THREAD

**Issue by kimrutherford** [pombase/website#2038]
V2 has a bunch of new features compared to v1 but I don't think there's a huge hurry to upgrade.

When we add v2 to PomBase, I think we should run v1 and v2 in parallel for a while to give the community time to get used to it.   (And time for us to fix any problems)

There's a bunch of demo instances here: https://jbrowse.org/jb2/demos/

It's now published which is what reminded me to make an issue: https://link.springer.com/article/10.1186/s13059-023-02914-z

**Comment by kimrutherford** [comment]
I'm taking the [JBrowse v2 paper](https://doi.org/10.1186/s13059-023-02914-z) on the flight with me.  :-)

**Comment by ValWood** [comment]
Woo Hoo.

**Comment by ValWood** [comment]
What are the new features? Anything snazzy?

**Comment by kimrutherford** [comment]
"Setting Up the JBrowse 2 Genome Browser":
https://currentprotocols.onlinelibrary.wiley.com/doi/pdf/10.1002/cpz1.1120

**Comment by kimrutherford** [comment]
I've been looking at getting JBrowse V2 set up as `pombase.org/jbrowse2/`.  It's not there yet.  :-)

My aim is to get the gene tracks up and running first and then work on the other tracks and features.

I initially wasn't sure that V2 supported embedding in gene pages, but there's a nice tutorial for it:
https://jbrowse.org/jb2/docs/tutorials/embed_linear_genome_view/01_introduction/

The tutorial also covers how to set up a default session. That will be helpful to us for users who are used to using JBrowse V1.  Without a default session we'll see a page like this: https://jbrowse.org/code/jb2/v2.18.0/

It looks like FlyBase, SGD, XenBase and RGD are still using JBrowse V1 so we're not the only ones.
The Alliance is using V2: https://www.alliancegenome.org/jbrowse2/

**Comment by ValWood** [comment]
Fantastic!

**Comment by kimrutherford** [comment]
> My aim is to get the gene tracks up and running first and then work on the other tracks and features.

First attempt:
https://desktop.kmr.nz/jbrowse2/?session=share-puR-5oWX0L&password=mfhN9
(I'm not sure that link will work)

![Image](https://github.com/user-attachments/assets/8143c298-3bbb-4b3f-b1fe-4c0c5cfa0ea4)

**Comment by kimrutherford** [comment]
JBrowse 2 is now updated each night with the rest of the site: https://www.pombase.org/jbrowse2/

It only has a gene track for now.  The next thing I'll do is split the track into forward and reverse strands.

**Comment by ValWood** [comment]
Looks flashy!

**Comment by ValWood** [comment]
I like the "move fast and break things" approach ;) or should it be "move fast and fix things"

**Comment by kimrutherford** [comment]
> I like the "move fast and break things" approach ;)

That's good because there could be a bit of that in the next few weeks.  :-)

I worked out what the problem was we JBrowse1.  I had made a typo in the load script.  The script is now fixed and conveniently I don't need to re-run the whole load to fix pombase.org.  I'm re-releasing now.

**Comment by kimrutherford** [comment]
> I don't need to re-run the whole load to fix pombase.org. I'm re-releasing now

Fixed and re-released.

**Comment by kimrutherford** [comment]
JBrowse2 v3 (confusing!) was released yesterday: https://jbrowse.org/jb2/blog/2025/01/29/v3.0.0-release/

We automatically use the latest version each nightly so https://www.pombase.org/jbrowse2/ is v3.0.0 this morning.

**Comment by kimrutherford** [comment]
We're getting there.  Tomorrow we'll have all the bigwig tracks in JBrowse 2 on pombase.org.  It still needs work but it will look something like this:

![Image](https://github.com/user-attachments/assets/b80bdb40-5e9b-4bee-9e22-a0062a04fd85)

**Comment by kimrutherford** [comment]
> Tomorrow we'll have all the bigwig tracks in JBrowse 2 on pombase.org. 

That worked.  There is still a bit of configuration to do.

I'll work on the other track types over the next week.

**Comment by kimrutherford** [comment]
I've configured the other track types now.  It didn't take as long as I thought.  The changes are on the main site:
https://www.pombase.org/jbrowse2/

All the tracks I've tried have worked but please let me know if you find anything not working.

There is still a lot of configuration to do but at least all the data is visible.

![Image](https://github.com/user-attachments/assets/4e48d5ab-0cb5-4db1-b2f4-7ec1aa11723f)

**Comment by kimrutherford** [comment]
> All the tracks I've tried have worked

I spoke too soon.  The "Consensus transcription factor DNA binding motifs" gives an error:
```
Error: invalid feature data, end less than start. end: 122207 start: 122211
```

The data file is: https://www.pombase.org/internal_datasets/TF_binding_motifs.bed.gz
Index: https://www.pombase.org/internal_datasets/TF_binding_motifs.bed.gz.tbi

Oddly the same data files work in JBrowse V1 so I wonder if it's just a configuration problem.

**Comment by kimrutherford** [comment]
> I spoke too soon. The "Consensus transcription factor DNA binding motifs" gives an error

That's the fixed now.  Turns out that the BED file for that track had dodgy formatting that JBrowse V1 coped with but V2 didn't like.

**Comment by ValWood** [comment]
Kim it looks great!  You could release it. That would be funny ;) 24 hours!

**Comment by kimrutherford** [comment]
> You could release it.

There are still a few bugs to fix.  Let's chat about releasing on Tuesday.

**Comment by ValWood** [comment]
OK, looks great though!

**Comment by kimrutherford** [comment]
> There are still a few bugs to fix.

I've fixed one of the problems: you can now type a gene name into the JBrowse 2 search box to go to that gene.  It doesn't work for systematic IDs yet.

**Comment by kimrutherford** [comment]
Jobs to do:
  - [ ] add links from feature in JBrowse to gene pages
  - [ ] colour features by type - use different colours for at least:
      - protein coding genes
      - pseudo-genes
      - RNA genes 
      - other (LTR, centromeres, low_complexity_region, ...)

**Comment by kimrutherford** [comment]
> add links from feature in JBrowse to gene pages

That's done now.  You need to click on the gene and then look in the details panel:

![Image](https://github.com/user-attachments/assets/ed30b577-c9c9-4760-9afe-adb34170b146)

**Comment by kimrutherford** [comment]
> colour features by type

That's done too:

![Image](https://github.com/user-attachments/assets/5fd81ad8-6467-4580-8e02-d54292230660)

**Comment by kimrutherford** [comment]
> you can now type a gene name into the JBrowse 2 search box to go to that gene.

That's broken again.  I'll fix it tomorrow.

**Comment by kimrutherford** [comment]
> That's broken again. I'll fix it tomorrow.

I can't work out the problem so I'll report it as a bug.

**Comment by kimrutherford** [comment]
> I can't work out the problem so I'll report it as a bug.

  - https://github.com/GMOD/jbrowse-components/issues/4814

**Comment by kimrutherford** [comment]
> I can't work out the problem so I'll report it as a bug.

I've implemented a work-around so the search box works now, even for "wtf22".

There are two more things I'd like to sort out:
  - [ ] link directly to a chromosome position from gene pages
  - [ ] default to showing chromosome 1 and the gene tracks

**Comment by kimrutherford** [comment]
The genes are now coloured by characterisation status (I hope!).  I've made the RNA genes grey.  Let me know if you see anything that's the wrong colour.

![Image](https://github.com/user-attachments/assets/6c62dc96-8fc7-49e6-9e74-d24348fea601)

**Comment by ValWood** [comment]
That looks correct!

**Comment by kimrutherford** [comment]
> There are two more things I'd like to sort out:

Any objections to me closing this issue and creating a new issue for the remaining JBrowse tasks/bugs?

**Comment by ValWood** [comment]
Not from me. It's just you talking to yourself and me occasionally saying "fantastic"  ;)

**Comment by kimrutherford** [comment]
OK, moved to:
 - pombase/website#2327
